### PR TITLE
Updated zip package

### PR DESCRIPTION
**Pull request by maryqyh**
Updated Zip source and test files. Reflected new functions and buffer changes in native file. Created FileTime class to replace java.nio import. Commented out ResourceLeakageDetector in test files due to different junit version.

### CODE REVIEW

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java [hunk 1]
```diff
@@ -24,11 +24,16 @@
 import java.util.zip.DataFormatException;
 import java.util.zip.Deflater;
 import java.util.zip.DeflaterInputStream;
-
-import junit.framework.TestCase;
 import libcore.io.Streams;
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java:28]: Could you follow tho conventions when we comment out lines due to J2ObjC-specific needs? Check other parts of jre_emul where we use block comments with a line "J2ObjC removed: [rationale]" as the first line. We can also note a TODO item for this.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java [hunk 2]
```diff
@@ -324,19 +344,20 @@ public void testSkip() throws IOException {
...
-        dis = new DeflaterInputStream(is);
-        assertEquals(23, dis.skip(Long.MAX_VALUE));
-        assertEquals(0, dis.available());
     }
 
     /**
      * DeflaterInputStream#DeflaterInputStream(InputStream)
      */
-    public void testDeflaterInputStreamInputStream() {
+//    @DisableResourceLeakageDetection(
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java:352]: Again, follow the J2ObjC convention for commenting lines out.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java [hunk 3]
```diff
@@ -26,10 +26,15 @@
...
+//import libcore.junit.junit3.TestCaseWithRules;
+//import libcore.junit.util.ResourceLeakageDetector;
+//import libcore.junit.util.ResourceLeakageDetector.DisableResourceLeakageDetection;
+import org.junit.Rule;
+import org.junit.rules.TestRule;
 
-import junit.framework.TestCase;
-
-public class DeflaterOutputStreamTest extends TestCase {
+public class DeflaterOutputStreamTest extends junit.framework.TestCase /* TestCaseWithRules */ {
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java:35]: Follow the convention for commenting out lines.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java [hunk 4]
```diff
@@ -169,19 +175,27 @@ public void test_ConstructorLjava_io_OutputStreamLjava_util_zip_DeflaterI()
         dos.write(byteArray);
         dos.close();
         f1.delete();
+        defl.end();
     }
 
     /**
      * java.util.zip.DeflaterOutputStream#close()
      */
+//    @DisableResourceLeakageDetection(
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java:184]: Here too.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java [hunk 5]
```diff
@@ -273,17 +286,22 @@ public void test_finish() throws Exception {
...
-        fos1.close();
         dos.close();
+        dos4.close();
         f1.delete();
     }
 
     /**
      * java.util.zip.DeflaterOutputStream#write(int)
      */
+//    @DisableResourceLeakageDetection(
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java:299]: And here.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java [hunk 6]
```diff
@@ -313,13 +331,25 @@ public void test_writeI() throws Exception {
...
+        } catch (IOException e) {
+        }
 
         f1.delete();
     }
 
     /**
      * java.util.zip.DeflaterOutputStream#write(byte[], int, int)
      */
+//    @DisableResourceLeakageDetection(
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java:347]: One more place.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java [hunk 7]
```diff
@@ -23,11 +23,15 @@
 import java.util.zip.DataFormatException;
 import java.util.zip.Deflater;
 import java.util.zip.Inflater;
-
-import junit.framework.TestCase;
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java:26]: Follow the convention for commenting out lines.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java [hunk 8]
```diff
@@ -23,11 +23,15 @@
...
+//import libcore.junit.util.ResourceLeakageDetector;
+import org.junit.Rule;
+import org.junit.rules.TestRule;
 import tests.support.resource.Support_Resources;
 
-public class DeflaterTest extends TestCase {
+public class DeflaterTest extends junit.framework.TestCase /* TestCaseWithRules */ {
+//    @Rule
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java:33]: Here too.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPInputStreamTest.java [hunk 9]
```diff
@@ -27,10 +27,17 @@
 import java.util.zip.Checksum;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.GZIPOutputStream;
-
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPInputStreamTest.java:30]: Follow the convention commenting out lines. Same below for the rest of the file.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPOutputStreamTest.java [hunk 10]
```diff
@@ -25,8 +25,14 @@
 import java.util.zip.Checksum;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.GZIPOutputStream;
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPOutputStreamTest.java:28]: Follow the convention commenting out lines. Same below for the rest of the file.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterInputStreamTest.java [hunk 11]
```diff
@@ -18,19 +18,22 @@
...
 import java.io.File;
 import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.zip.DeflaterOutputStream;
 import java.util.zip.Inflater;
 import java.util.zip.InflaterInputStream;
-
-import junit.framework.TestCase;
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterInputStreamTest.java:28]: Follow the convention commenting out lines. Same below for the rest of the file.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterTest.java [hunk 12]
```diff
@@ -27,9 +27,16 @@
 import java.util.zip.DeflaterOutputStream;
 import java.util.zip.Inflater;
 import java.util.zip.ZipException;
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterTest.java:30]: Follow the convention commenting out lines. Same below for the rest of the file.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java [hunk 13]
```diff
@@ -20,13 +20,22 @@
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+//import java.nio.file.attribute.FileTime;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java:23]: Probably we need to add a "J2ObjC removed" comment above and follow the convention.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java [hunk 14]
```diff
@@ -20,13 +20,22 @@
...
+import java.util.zip.ZipUtils.FileTime; /* J2ObjC Added: FileTime class to avoid java.nio.file.attribute.FileTime */
 import java.util.TimeZone;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipFile;
 import libcore.io.Streams;
+//import libcore.junit.junit3.TestCaseWithRules;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java:29]: Same here and for the rest of the file.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipOutputStream.java [hunk 15]
```diff
@@ -44,9 +45,24 @@
...
+     * Whether to use ZIP64 for zip files with more than 64k entries.
+     * Until ZIP64 support in zip implementations is ubiquitous, this
+     * system property allows the creation of zip files which can be
+     * read by legacy zip implementations which tolerate "incorrect"
+     * total entry count fields, such as the ones in jdk6, and even
+     * some in jdk7.
+     */
+    // Android-changed: Always allow use of Zip64.
+    private static final boolean inhibitZip64 = false;
+    //  Boolean.parseBoolean(
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipOutputStream.java:58]: Is this J2ObjC-specific for already there in libcore?

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 16]
```diff
@@ -0,0 +1,211 @@
...
+
+//import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public
+class ZipUtils {
+
+    /**
+     * J2ObjC added: avoids using java.nio.file.attribute.FileTime,
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java:36]: Could you add unit test cases to ZipUtilsTest for this class?

**Comment by maryqyh** [comment]
Should I submit a PR for just Makefiles, a PR for all updated source files, and another for all test files?

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 17]
```diff
@@ -0,0 +1,211 @@
...
+            if (unit.equals(TimeUnit.MICROSECONDS)) {
+                return new FileTime( value / 1000);
+            }
+            if (unit.equals(TimeUnit.MILLISECONDS)) {
+                return new FileTime(value);
+            }
+            if (unit.equals(TimeUnit.SECONDS)) {
+                return new FileTime(1000 * value);
+            }
+            return new FileTime(0);
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java:57]: I feel that switch-case may be more appropriate here. How do you make sure that these are the only `TimeUnit` enums the package is using? Same question below.

**Comment by lukhnos** [comment]
> Should I submit a PR for just Makefiles, a PR for all updated source files, and another for all test files?

Makefile changes need to go with the source file changes (otherwise it would break the build). It's just the IntelliJ project file that we want to update separately (as a separate PR if warranted, that is).

File: .idea/workspace.xml [hunk 18]
```diff
@@ -1,7 +1,39 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (tomball) [inline .idea/workspace.xml:1]: Please revert this file. If we need to update .idea/workspace.xml, it should be done in a separate commit.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 19]
```diff
@@ -0,0 +1,211 @@
...
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util.zip;
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java:26]: We cannot change java.util.zip's public API. Please change this class's package to com.google.j2objc.util, in jre_emul/Classes/com/google/j2objc/util/. I think in IntelliJ you can just drag this file into that directory, and it will automatically refactor the class and references to it.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 20]
```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2013, 2015, Oracle and/or its affiliates. All rights reserved.
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java:2]: Change the copyright to our Apache license, since you wrote it and not Oracle engineers. The easiest way to do so is to first move the class to com.google.j2objc.util (see below comment), and then copy the copyright comment from any of that package's source files.

File: jre_emul/jre_sources.mk [hunk 21]
```diff
@@ -1680,6 +1680,7 @@ JAVA_PRIVATE_SOURCES_ZIP = \
   java/util/zip/ZStreamRef.java \
   java/util/zip/ZipCoder.java \
   java/util/zip/ZipConstants64.java \
+  java/util/zip/ZipUtils.java \
```
Comment (tomball) [inline jre_emul/jre_sources.mk:1683]: Change to com/google/j2objc/util/ZipUtils.java and move to its alphabetical location, per above comments.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 22]
```diff
@@ -0,0 +1,211 @@
...
+package java.util.zip;
+
+//import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public
+class ZipUtils {
+
+    /**
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java:35]: Please reformat for Google standards (2 space indents), which is easy if you have the google-java-format IntelliJ plugin. You were correct to use the same formatting as the rest of the package, but it's being moved to a Google-authored package now.

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java (hunk 1 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java]: added comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java (hunk 2 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterInputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java (hunk 3 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java (hunk 4 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java (hunk 5 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java (hunk 13 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java (hunk 14 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/ZipEntryTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipOutputStream.java (hunk 15 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipOutputStream.java:58]: this is already there in libcore, and no changes are made specifically for J2ObjC

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 19 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: changed file location and relevant import lines

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 20 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: copied the copyright from other files in the same package

File: jre_emul/jre_sources.mk (hunk 21 above)
Comment (maryqyh) [reply jre_emul/jre_sources.mk]: changed file path

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 22 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: changed style for the file according to other files in the same package

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java (hunk 6 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterOutputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java (hunk 7 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPInputStreamTest.java (hunk 9 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPInputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPOutputStreamTest.java (hunk 10 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/GZIPOutputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterInputStreamTest.java (hunk 11 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterInputStreamTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterTest.java (hunk 12 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/InflaterTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java (hunk 8 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/harmony-tests/src/test/java/org/apache/harmony/tests/java/util/zip/DeflaterTest.java]: added J2ObjC comments

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 22 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: reformatted FileTime.java

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 17 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: I changed the implementation into switch-case and included all TimeUnit enums.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 16 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: added FileTimeTest.java in misc_tests/com/google/j2objc/nio/file/attribute

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 23]
```diff
@@ -0,0 +1,211 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util.zip;
+
+//import java.nio.file.attribute.FileTime;
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: Change to:

// j2objc: use platform-specific FileTime.
// import java.nio.file.attribute.FileTime;
import com.google.j2objc.nio.file.attribute.FileTime;

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java [hunk 24]
```diff
@@ -0,0 +1,211 @@
...
+import java.util.concurrent.TimeUnit;
+
+public
+class ZipUtils {
+
+    /**
+     * J2ObjC added: avoids using java.nio.file.attribute.FileTime,
+     * instead implements FileTime class with relevant methods.
+     */
+    public static class FileTime implements Comparable<FileTime>{
```
Comment (tomball) [inline jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: As we discussed, please move this inner class to a new source file in jre_emul/Classes/com/google/j2objc/nio/file/attribute/FileTime.java, which will use the Apache license and may include you as its @author in the class doc-comment. This reduces the delta on ZipUtils.java to just the changed import statement above.

File: jre_emul/jre_sources.mk (hunk 21 above)
Comment (tomball) [inline jre_emul/jre_sources.mk:1684]: Add com/google/j2objc/nio/file/attribute/FileTime.java to JAVA_PRIVATE_SOURCES_ZIP. We'll move it as needed if other classes reference it.

File: jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java (hunk 24 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/zip/ZipUtils.java]: I think I already moved this class and changed corresponding licenses in my most recent commit.

File: jre_emul/jre_sources.mk (hunk 21 above)
Comment (maryqyh) [reply jre_emul/jre_sources.mk:1684]: I've added this file in JAVA_PRIVATE_SOURCES_ZIP in my commit ccb13028f3ea1a603ba60a7cab486e33fddfdc0f on line 1679.

**Comment by tomball** [comment]
Superceded by https://github.com/google/j2objc/pull/1335
